subgraphs/burrow/src/handlers/liquidate.ts

```diff
@@ -0,0 +1,402 @@
+
+  /* -------------------------------------------------------------------------- */
+  /*                                 Liquidatee                                 */
+  /* -------------------------------------------------------------------------- */
+  const liquidation_account_id = data.get("liquidation_account_id");
+  if (!liquidation_account_id) {
+    log.info("{} data not found", ["liquidation_account_id"]);
+    return;
+  }
+  const liquidatee = getOrCreateAccount(liquidation_account_id.toString());
+  if (liquidatee.liquidationCount == 0) {
+    protocol.cumulativeUniqueLiquidatees =
+      protocol.cumulativeUniqueLiquidatees + 1;
+  }
+  if (liquidatee._last_active_timestamp.lt(usageDaily.timestamp)) {
+    usageDaily.dailyActiveLiquidatees += 1;
+  }
+  liquidatee.liquidationCount += 1;
+  liq.liquidatee = liquidatee.id;
+
+  /* -------------------------------------------------------------------------- */
+  /*                              Collateral Amount                             */
+  /* -------------------------------------------------------------------------- */
+  const collateral_sum = data.get("collateral_sum");
+  if (!collateral_sum) {
+    log.info("{} data not found", ["collateral_sum"]);
+    return;
+  }
+
+  const collateral_sum_value = BigDecimal.fromString(collateral_sum.toString());
+  liq.amountUSD = collateral_sum_value;
+
+  /* -------------------------------------------------------------------------- */
+  /*                                Repaid Amount                               */
+  /* -------------------------------------------------------------------------- */
+  const repaid_sum = data.get("repaid_sum");
+  if (!repaid_sum) {
+    log.info("{} data not found", ["repaid_sum"]);
+    return;
+  }
+  const repaid_sum_value = BigDecimal.fromString(repaid_sum.toString());
+  liq.profitUSD = liq.amountUSD.minus(repaid_sum_value);
+
+  // finding token_in, token_in_amount, token_out and token_out_amount
+  // TOKEN_IN: borrowed token
+  // TOKEN_OUT: collateral token
+  let token_in: string | null = null,
+    token_out: string | null = null;
+  let token_in_amount: string | null = null,
+    token_out_amount: string | null = null;
+  if (args) {
+    let msg = args.get("msg");
+    if (!msg) {
+      log.info("LIQ::Msg not found", []);
+      return;
+    }
+    log.info("LIQ::MSG {}", [msg.toString()]);
+    msg = json.fromString(msg.toString());
+    const exec = msg.toObject().get("Execute");
+    if (!exec) {
+      log.info("LIQ::Execute not found", []);
+      return;
+    }
+
+    if (exec.kind != JSONValueKind.OBJECT) return;
+    const actions = exec.toObject().get("actions");
+    if (!actions) {
+      log.info("LIQ::Actions not found", []);
+      return;
+    }
+
+    if (actions.kind != JSONValueKind.ARRAY) return;
+    const actionsArr = actions.toArray();
+
+    for (let i = 0; i < actionsArr.length; i++) {
+      if (actionsArr[i].kind == JSONValueKind.OBJECT) {
```

Comment:
this seems excessively complex. To fix that early returns (breaks in a for loop) will make it easier to read

Reply:
will do